packages/mdc-snackbar/index.js

```diff
@@ -21,4 +21,192 @@
+   *     client-specific values may also be used if desired.
+   */
+  close(reason = '') {
+    if (!this.isOpen) {
+      return;
+    }
+    this.deregisterDocumentKeyDownHandler_(this.handleDocumentKeyDown_);
+    this.foundation_.close(reason);
+  }
+
+  /**
+   * @return {!MDCSnackbarFoundation}
+   */
+  getDefaultFoundation() {
+    /* eslint brace-style: "off" */
+    return new MDCSnackbarFoundation({
+      hasClass: (className) => this.root_.classList.contains(className),
+      addClass: (className) => this.root_.classList.add(className),
+      removeClass: (className) => this.root_.classList.remove(className),
+
+      notifyOpening: () => this.emit(MDCSnackbarFoundation.strings.OPENING_EVENT),
+      notifyOpened: () => this.emit(MDCSnackbarFoundation.strings.OPENED_EVENT),
+      notifyClosing: (reason) => this.emit(MDCSnackbarFoundation.strings.CLOSING_EVENT, {reason}),
+      notifyClosed: (reason) => this.emit(MDCSnackbarFoundation.strings.CLOSED_EVENT, {reason}),
+    });
+  }
+
+  /**
+   * @return {number}
+   */
+  get timeoutMs() {
+    return this.foundation_.timeoutMs;
+  }
+
+  /**
+   * @param {number} timeoutMs
+   */
+  set timeoutMs(timeoutMs) {
+    this.foundation_.timeoutMs = timeoutMs;
+  }
+
+  /**
+   * @return {boolean}
+   */
+  get closeOnEscape() {
+    return this.foundation_.closeOnEscape;
+  }
+
+  /**
+   * @param {boolean} closeOnEscape
+   */
+  set closeOnEscape(closeOnEscape) {
+    this.foundation_.closeOnEscape = closeOnEscape;
+  }
+
+  /**
+   * @return {boolean}
+   */
+  get isOpen() {
+    return this.foundation_.isOpen;
+  }
+
+  /**
+   * @return {string}
+   */
+  get labelText() {
+    const {LABEL_SELECTOR} = MDCSnackbarFoundation.strings;
+    return this.root_.querySelector(LABEL_SELECTOR).textContent;
+  }
+
+  /**
+   * @param {string} labelText
+   */
+  set labelText(labelText) {
+    const {LABEL_SELECTOR} = MDCSnackbarFoundation.strings;
+    this.root_.querySelector(LABEL_SELECTOR).textContent = labelText;
+  }
+
+  /**
+   * @return {string}
+   */
+  get actionButtonText() {
+    const {ACTION_BUTTON_SELECTOR} = MDCSnackbarFoundation.strings;
+    return this.root_.querySelector(ACTION_BUTTON_SELECTOR).textContent;
+  }
+
+  /**
+   * @param {string} actionButtonText
+   */
+  set actionButtonText(actionButtonText) {
+    const {ACTION_BUTTON_SELECTOR} = MDCSnackbarFoundation.strings;
+    this.root_.querySelector(ACTION_BUTTON_SELECTOR).textContent = actionButtonText;
+  }
+
+  /**
+   * @return {!HTMLElement}
+   * @private
+   */
+  get surfaceEl_() {
+    const {SURFACE_SELECTOR} = MDCSnackbarFoundation.strings;
+    return this.root_.querySelector(SURFACE_SELECTOR);
+  }
+
+  /**
+   * @param {!EventListener} handler
+   * @private
+   */
+  registerTransitionEndHandler_(handler) {
+    this.root_.addEventListener(getCorrectEventName(window, 'transitionend'), handler);
```

Comment:
You can use `this.listen` in place of `this.root_.addEventListener` and `this.unlisten` in place of `this.root_.removeEventListener`

Reply:
Done